Support selecting a grid row by partial map of column, text values

labkey-chrisj opened the pull request:
#### Rationale
This adds support for selecting a row in a QueryGrid based on values it might have in more than one column.
The grid already supports finding a row by passing in a partialMap (of column, text).

#### Related Pull Requests
https://github.com/LabKey/biologics/pull/697

#### Changes
New overload for `selectRow()`

labkey-chrisj committed:
Support selecting a grid row by partial map of column, text values

labkey-kevink commented on src/org/labkey/test/components/glassLibrary/grids/ResponsiveGrid.java:
```diff
@@ -141,6 +141,18 @@ public ResponsiveGrid selectRow(int index, boolean checked)
+     * @param checked       the desired checkbox state
+     * @return
+     */
+    public ResponsiveGrid selectRow(Map partialMap, boolean checked)
```
You should include the generic types for the Map.

```suggestion
    public ResponsiveGrid selectRow(Map<String, String> partialMap, boolean checked)
```

Also the `selectRow(String columnLabel, String text, boolean checked)` method validates the checkbox count after selecting.  Do you need to do that here too?

labkey-chrisj replied:
It's probably a good idea so I will

labkey-chrisj committed:
Merge remote-tracking branch 'origin/develop' into fb_smIntegrationSampleDeleteTest

labkey-chrisj committed:
CR feedback
- include generic map params
- verify expected selection counts when setting row select state
